querydsl-libraries/querydsl-core/src/main/java/com/querydsl/core/group/GroupByBuilder.java

```diff
@@ -287,8 +287,7 @@ public <V, RES extends Collection<V>> ResultTransformer<RES> collection(
       Supplier<RES> resultFactory, FactoryExpression<V> expression) {
     final FactoryExpression<V> transformation = FactoryExpressionUtils.wrap(expression);
     var args = transformation.getArgs();
-    return new GroupByGenericCollection<>(
-        resultFactory, key, args.toArray(new Expression<?>[args.size()])) {
+    return new GroupByGenericCollection<>(resultFactory, key, args.toArray(new Expression<?>[0])) {
```

Comment:
Since we do have the necessary stay length, I would say to keep it. 

There is no gain on using this. 

Or, switch to `Expression[]:new`

Reply:
https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size
https://shipilev.net/blog/2016/arrays-wisdom-ancients/
https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html